backend/src/routes/weaviate.ts

```diff
@@ -45,15 +54,61 @@ weaviateRouter.get("/", async (_req, res) => {
 weaviateRouter.get("/test", async (_req, res) => {
   const result = await client.graphql
     .get()
-    .withClassName('ResearchPaper')
+    .withClassName("ResearchPaper")
     .withHybrid({
-      query: 'software',
+      query: "software",
     })
     .withLimit(3)
-    .withFields('title content')
+    .withFields("title content")
     .do();
 
   res.status(201).json({ response: JSON.stringify(result, null, 2) });
 });
 
+weaviateRouter.post("/search", async (req, res) => {
+  const inputString = req.body.text;
+  console.log(inputString);
+
+  if (!inputString) {
+    res.status(400).json({ error: "text is required in request body" });
+  }
+
+  const promptText = `You are an entity extractor now. Extract entities from the following text: '${inputString}'`;
+  try {
+    const response = await axios.post(
+      OPEN_AI_API_URL,
+      {
+        model: "gpt-3.5-turbo",
+        messages: [{ role: "system", content: promptText }],
+        temperature: 0.7,
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${OPEN_AI_APIKEY}`,
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const entities = response.data.choices[0].message.content.trim();
+    const result = await client.graphql
+      .get()
+      .withClassName("ResearchPaper")
+      .withGenerate({
+        groupedTask: inputString,
+        groupedProperties: ["content", "title"],
+      })
+      .withNearText({
+        concepts: [entities],
+      })
+      .withFields("chunk_index")
+      .withLimit(4)
+      .do();
+
+    // const { userInput } = req.body;
+    res.status(201).json({ response: JSON.stringify(result, null, 2) });
```

Comment:
To make Frontend developer life easier, I would wrap the Weaviate result for some nicer format.

Currently the response is like this:

<img width="1460" alt="Screenshot 2023-10-31 at 21 17 52" src="https://github.com/asianomainen/rag-a-tag/assets/40372306/1fd0b707-3f20-4cef-ac51-a57b30843ec3">

Maybe to have a return object with values:

```
{
     "gptResponse": "blaa blaa blaa gpt generated response" <-- groupedResult 
     "references": [<chunk_index_value>, <chunk_index_value>] <-- array of used weaviate document titles (chunk_index). chunk_index actually quite bad name
}
```